github/migrations_source_import.go

```diff
@@ -0,0 +1,313 @@
+	//         total number of raw commits that will be imported) and
+	//         percent (0 - 100, the current progress through the import).
+	//     mapping - the "rewrite" step of the import is in progress. This
+	//         is where SVN branches are converted to Git branches, and
+	//         where author updates are applied. The import progress
+	//         response does not include progress information.
+	//     pushing - the "push" step of the import is in progress. This is
+	//         where the importer updates the repository on GitHub. The
+	//         import progress response will include push_percent, which is
+	//         the percent value reported by git push when it is "Writing
+	//         objects".
+	//     complete - the import is complete, and the repository is ready
+	//         on GitHub.
+	//
+	// If there are problems, you will see one of these in the status field:
+	//
+	//     auth_failed - the import requires authentication in order to
+	//         connect to the original repository. Make an "Update Existing
+	//         Import" request, and include vcs_username and vcs_password.
+	//     error - the import encountered an error. The import progress
+	//         response will include the failed_step and an error message.
+	//         Contact GitHub support for more information.
+	//     detection_needs_auth - the importer requires authentication for
+	//         the originating repository to continue detection. Make an
+	//         "Update Existing Import" request, and include vcs_username
+	//         and vcs_password.
+	//     detection_found_nothing - the importer didn't recognize any
+	//         source control at the URL.
+	//     detection_found_multiple - the importer found several projects
+	//         or repositories at the provided URL. When this is the case,
+	//         the Import Progress response will also include a
+	//         project_choices field with the possible project choices as
+	//         values. Make an "Update Existing Import" request, and
+	//         include vcs and (if applicable) tfvc_project.
+	Status        *string  `json:"status,omitempty"`
+	CommitCount   *int     `json:"commit_count,omitempty"`
+	StatusText    *string  `json:"status_text,omitempty"`
+	AuthorsCount  *int     `json:"authors_count,omitempty"`
+	ImportPercent *float64 `json:"import_percent,omitempty"`
+	URL           *string  `json:"url,omitempty"`
+	HTMLURL       *string  `json:"html_url,omitempty"`
+	AuthorsURL    *string  `json:"authors_url,omitempty"`
+	RepositoryURL *string  `json:"repository_url,omitempty"`
+	Message       *string  `json:"message,omitempty"`
+	FailedStep    *string  `json:"failed_step,omitempty"`
+}
+
+func (i Import) String() string {
+	return Stringify(i)
+}
+
+// SourceImportAuthor identifies an author imported from a source repository.
+type SourceImportAuthor struct {
+	ID         *int    `json:"id,omitempty"`
+	RemoteID   *string `json:"remote_id,omitempty"`
```

Comment:
You might want to add a comment pointing to the original JSON GitHub docs.
Is it possible that `RemoteID` is an `*int`?
I remember being surprised before by the GitHub API where I expected a `string` but got an `int`.